scripts/convert_config.py

```diff
@@ -0,0 +1,460 @@
+
+    parser.add_argument('--component', help='Select the components for which '
+                        'configuration files should be generated. If not '
+                        'provided, generate configuration files for all '
+                        'components. Can be provided multiple times',
+                        default=[],
+                        action='append',
+                        nargs='+')
+
+    parser.add_argument('--debug', help='Print resulting config after each '
+                        'applied mapping in JSON format', default=False,
+                        action='store_true')
+
+    parser.add_argument('--input', help='Input keylime configuration file to '
+                        'process. If not provided, it tries to use the '
+                        'installed Keylime configuration')
+
+    parser.add_argument('--out', help='Output directory where to put the '
+                        'generated files. Default is current directory', default='.')
+
+    parser.add_argument('--mapping', help='If provided, then only the provided '
+                        'mapping file will be used. Otherwise, mappings for '
+                        'each version found in the "templates" directory will '
+                        'be applied. The file must be a JSON file containing '
+                        'the mapping for the new configuration options names '
+                        'to dictionaries containing the section and option '
+                        'name from the old configuration file, and the default '
+                        'value to use in case the option is missing in the '
+                        'input configuration.',
+                        default=None)
+
+    parser.add_argument('--templates', help='Path to the directory containing '
+                        'the templates for the configuration files. If not '
+                        'provided "./templates" is used', default=None)
+
+    parser.add_argument('--version', help='Target version for the output '
+                        'configuration files', default=None)
+
+    args = parser.parse_args()
+
+    if not os.path.exists(args.out):
+        raise Exception(f"Output directory {args.out} does not exist")
+
+    if not os.path.isdir(args.out):
+        raise Exception(f"File {args.out} is not a directory")
+
+    if args.version and not is_version(args.version):
+        raise Exception(f"Invalid version {args.version} specified in --version: "
+                        f"Expected 'MAJOR.MINOR' format")
+
+    if args.templates:
+        if not os.path.exists(args.templates):
+            raise Exception(
+                f"Templates directory {args.template} does not exist")
+
+        if not os.path.isdir(args.templates):
+            raise Exception(f"File {args.templates} is not a directory")
+
+        templates = args.templates
+    else:
+        templates = "./templates"
```

Comment:
If you add `default="./templates"` in the `add_argument`, instead of `None`, you can skip this line and also will run the validation (check the the patch exists) in all cases

Reply:
Thanks, I'll change the code.